### DISCUSSION TITLE
Possible to add object/event attributes over API?

### DISCUSSION THREAD

**Discussion by MisterEcks** [blakeblackshear/frigate#10171] [Ask A Question]
### What is your question:

I didn't see a way to add object attributes via API and am wondering if this is possible. Currently I am using the sublabel field to augment event/objects with relevant data, but this is tricky for cars with Amazon/FedEx/Ups detection enabled as it also uses and overwrites the sublabel field. I am using the sublabel to hold the ALPR result, vehicle direction, and vehicle name if known using delimiters. Having these as object attributes and not fighting over the sublabel would be better in my case.

**Comment by NickM-27** [comment]
Currently this is not possible

**Comment by glossyio** [comment]
I have a very similar use case with detecting vehicle direction and speeds, and I would like to attach these to Events in Frigate, so they can appear on the Frigate interface next to the events.  However, the speeds do not meet the subLabelScore requirements of <=1.0 and >=0.  

My thought then would be to create a separate database of these "external attributes" + `event_id` and join anything else I need from Frigate, creating an external page to view all that relevant data together. Would you concur?  In that case, I have a low priority feature request about adding arbitrary sub label data.

**Reply by NickM-27** [reply]
This isn't what sub label is designed for.

Frigate is also currently having the entire UI rewritten and it won't really fit your use case of showing this information. I think an external UI is best.

**Reply by glossyio** [reply]
Totally understand! I should really think of `sublabel` as a subordinate object label (and score) to the primary object label in the event; e.g. a license plate (and score) for a car or a face (and score) to a person.  
Thanks for the consideration.
